qutebrowser/browser/webkit/network/pac.py

```diff
@@ -0,0 +1,529 @@
+        carry_b = 0
+        for a, val, b, r in zip(avals, vals, bvals, ranges):
+            if a is not None:
+                a_c = a + carry_a
+                val_c = val + carry_val
+                b_c = b + carry_b
+                if a_c < val_c < b_c:
+                    return True
+                elif not a_c <= val_c <= b_c:
+                    return False
+                carry_a += a * r
+                carry_val += val * r
+                carry_b += b * r
+        return True
+
+    @_js_slot(QVariant)
+    def weekdayRange(self, args):
+        """Test is current date is between two week days.
+
+        Only the first parameter is mandatory. Either the second, the third,
+        or both may be left out.
+
+        If only one parameter is present, the function returns a value of
+        true on the weekday that the parameter represents. If the string
+        "GMT" is specified as a second parameter, times are taken to be in
+        GMT. Otherwise, they are assumed to be in the local timezone.
+
+        If both wd1 and wd1 are defined, the condition is true if the
+        current weekday is in between those two ordered weekdays. Bounds are
+        inclusive, but the bounds are ordered. If the "GMT" parameter is
+        specified, times are taken to be in GMT. Otherwise, the local
+        timezone is used.
+        """
+        weekdays = ("MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN")
+
+        # Expects list of all arguments because the function has different
+        # modes expecting different number of them.
+        if isinstance(args, QJSValue):
+            args = args.toVariant()
+        wd1 = args[0]
+        wd2 = wd1
+        gmt = args[-1] == "GMT"
+        if gmt:
+            args = args[:-1]
+
+        if len(args) == 2:
+            wd2 = args[1]
+        elif len(args) != 1:
+            raise EvalProxyError("Invalid number of arguments")
+
+        avals = (weekdays.index(wd1),)
+        bvals = (weekdays.index(wd2),)
+        d = self._get_dt(gmt).date()
+        val = (d.dayOfWeek() - 1,)
+
+        return self._in_range(avals, val, bvals, (1,))
+
+    def _date_range(self, args1, args2, gmt):
+        """Test if current date is between two date points.
+
+        Args:
+            args1, args2: two dictionaries with optional "year", "month" and
+            "day".
+            gmt: Should the UTC time be used.
+        """
+        def _to_tuple(args):
+            months = ("JAN", "FEB", "MAR", "APR", "MAY", "JUN", "JUL", "AUG",
+                      "SEP", "OCT", "NOV", "DEC")
+
+            if "month" in args:
+                m = months.index(args["month"])
+            else:
+                m = None
+
+            return (args.get("year"), m, args.get("day"))
+
+        ranges = (31 * 12, 31, 1)
+
+        d = self._get_dt(gmt).date()
+        val = (d.year(), d.month() - 1, d.day())
+
+        return self._in_range(_to_tuple(args1), val, _to_tuple(args2), ranges)
```

Comment:
Unfortunately this won't do because those values may be partly absent. For example, we may want to just check if day 20 is between 15 and 30, or if 16/04 is between 20/05 and 21/06.